pkg/build/build_implementation.go

```diff
@@ -138,7 +138,8 @@ func (di *defaultBuildImplementation) GenerateImageSBOM(o *options.Options, ic *
 		return nil
 	}
 
-	s := newSBOM(di.workdirFS, o, ic)
+	archWorkdirFS := apkfs.DirFS(o.WorkDir, apkfs.WithCreateDir(true))
+	s := newSBOM(archWorkdirFS, o, ic)
```

Comment:
This part confuses me. Is it trying to read the workdir? Or the tarball? Or is it making a new one?

The build process builds the entire arch-specific image layout, then wraps it up in a tarGZ layer:

```go
	for _, arch := range archs {
		arch := arch
		// working directory for this architecture
		wd := filepath.Join(workDir, arch.ToAPK())
		bc, err := build.New(wd, opts...)
		if err != nil {
			return err
		}

		// we do not generate SBOMs for each arch, only possibly for final image
		bc.Options.SBOMFormats = []string{}
		bc.Options.WantSBOM = false
		bc.ImageConfiguration.Archs = archs

		errg.Go(func() error {
			bc.Options.Arch = arch
			bc.Options.WorkDir = wd

			if err := bc.Refresh(); err != nil {
				return fmt.Errorf("failed to update build context for %q: %w", arch, err)
			}

			layerTarGZ, err := bc.BuildLayer()
```

Trying to rebuild an `apko.FullFS` based on that working directory might, or might not, work, depending on the semantics of the underlying filesystem. On the other hand, the layerTGZ definitely has everything correct.

Can we switch this? Can we either:

* generate the SBoM from the arch-specific layerTGZ? (might need to expand into a tempdir first); OR
* generate the SBoM after each arch-specific image is built, while we still have the original `FullFS` in hand?

I think the latter would work better.

I am happy to fix this, if you can give me exactly how to reproduce it.

Reply:
Yes, lets refactor it to use the original FullFS.  I think we should do some refactoring of the SBOM analysis engine anyway.  cc @luhring who probably has opinions.

Happy to hop on a gmeet in a few hours when people are awake to flesh this out.

Reply:
Yeah happy to jump on a call. I think I'm missing some context but would love to help out here